### ISSUE TITLE
Nextcloud snap and external storage

### ISSUE THREAD

**Issue by dln949** [nextcloud-snap/nextcloud-snap#2332]
Nextcloud snap 25.0.3 running on Ubuntu 22.04 (wayland).  

Not sure if this is a Nextcloud external storage issue, or a Nextcloud snap issue, I thought I'd ask here first.  

I ran this command:  sudo snap connect nextcloud:removable-media  (I assume I was supposed to literally type "removable-media" in that terminal command.)  

On the computer on which Nextcloud is installed, I plugged in a USB external storage.  It is mounted at /media/myuser/dir1.  I set the recurseive permissions on that mounted drive to be 777.  Since I am the one who plugged the drive in, the owner is _myuser_, not root.  

In Nextcloud Administration, under "External storage", I added this storage:  

"Folder name" = dir1
"External storage" = "Local"
"Authentication" = "None"
"Configuration" = /media/myuser/dir1
"Available for" = _MyNextcloudUser_, and admin(group)

In "Advanced Settings" I chose "Enable previews"

And then I clicked on the checkmark to "Save".  

I keep getting the red error symbol to "Click to recheck the configuration".  And when I go to "Files", the dir1 folder appears, but if I click on it, I get the error message, "External mount error / ! There was an error with message:  Empty response from the server.  Do you want to review mount point config in admin settings page?"  There followed two buttons to click, either "No" or "Yes".  But all clicking on the "Yes" button did was take me back to the "External storage" settings page that I had just been on.  

At this point I can't figure out next steps to determine what is wrong.

**Comment by scubamuc** [comment]
what you are describing is not really an issue, its a security design feature of Snaps called confinement. your external USB storage mus be owned and accessed only by root and should be mounted in `/mnt` or `/media`.

find UUID of the storage device:
`sudo blkid`

mount your external USB storage in `/etc/fstab` (option nofail allows headless reboot):
`UUID=xxxyyyttt555666777  /media/dir1  ext4 defaults,nofail  0  2`

there is a way to [bypass confinement](https://github.com/scubamuc/scubamuc.github.io/blob/scubamuc/wiki-md/NEXTCLOUD.snap--snap-confinement.md) but thats defeating the security aspect.

**Comment by dln949** [comment]
Thank you for clearing up my thinking, I was misunderstanding how things need to be set up.  It seems to be working now.  

Now I just need to figure out the bewildering (to me) options for the Backup application......

**Comment by scubamuc** [comment]
yip, the backup app is bewildring... try snap-backup istead. this can be scripted and automated via crontab:
https://github.com/scubamuc/scubamuc.github.io/blob/scubamuc/Readme.md#32-nextcloud-snap-snapshot--restore

**Comment by dln949** [comment]
Thanks very much for the two references!!!!  I do have a few questions:  

1. Do I understand the snapshot correctly, it DOES include copying the data on my Nextcloud server, such as family photos, family movies, other files we've saved, etc.?  
2. I'm not clear on the distinction between export and snapshot.  When should I use which one?  
3. In the snapshot script, what time of day does the script start?
4. Which day will be the first day this script runs?  
5. Once the snapshot script starts, I assume Nextcloud is unavailable until Nextcloud snap starts again - correct?  
6. Currently with all the large family movies and photos I've loaded into Nextcloud, the computer Nextcloud is on is about 70% full.  That won't present a problem for creating (or restoring) the snapshot?  It will place the snapshot on the external media I have mounted in fstab?  
7. My Nextcloud is currently about 700GB.  How large will the snapshot file be?  
8. How can I estimate how long it will take the snapshot script to run (and keep Nextcloud temporarily unavailable)?  

Thanks much!
